contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol

```diff
@@ -0,0 +1,497 @@
+        )
+        external
+        override
+        onlyReporters
+        inStatus(_queryId, Witnet.QueryStatus.Posted)
+    {
+        _reportResult(_queryId, block.number, _witnetProof, _witnetResult);
+    }
+
+    /// Reports the Witnet-provided result to a previously posted request.
+    /// @dev Fails if:
+    /// @dev   - called from unauthorized address;
+    /// @dev   - the `_queryId` is not in 'Posted' status.
+    /// @dev   - provided `_witnetProof` is zero;
+    /// @dev   - length of provided `_witnetValue` is zero.
+    /// @param _queryId The unique identifier of the request.
+    /// @param _witnetEpoch of the solving tally transaction in Witnet.
+    /// @param _witnetProof of the solving tally transaction in Witnet.
+    /// @param _witnetResult The result itself as bytes.
+    function reportResult(
+            uint256 _queryId,
+            uint256 _witnetEpoch,
+            bytes32 _witnetProof,
+            bytes calldata _witnetResult
+        )
+        external
+        override
+        onlyReporters
+        inStatus(_queryId, Witnet.QueryStatus.Posted)
+    {
+        _reportResult(_queryId, _witnetEpoch, _witnetProof, _witnetResult);
+    }
+    
+
+    // ================================================================================================================
+    // --- Full implementation of 'IWitnetRequestBoardRequestor' -------------------------------------------------------------
+
+    /// Retrieves Witnet-proved result to a previously posted request, removing the whole query from storage.
+    /// @dev Fails if the `_queryId` is not in 'Reported' status, or called from an address different to
+    /// @dev the one that actually posted the given request.
+    /// @param _queryId The unique identifier of a previously posted request.
+    function destroyResult(uint256 _queryId)
+        public
+        virtual override
+        inStatus(_queryId, Witnet.QueryStatus.Reported)
+        returns (bytes memory _result)
+    {
+        require(
+            msg.sender == _getRequestData(_queryId).requestor,
+            "WitnetRequestBoardTrustableBase: only requestor"
+        );
+        Witnet.Response storage _response = _getResponseData(_queryId);
+        _result = _response.witnetResult;
+        delete _state().queries[_queryId];
+        emit DestroyedResult(_queryId, msg.sender);
+    }
+
+    /// Requests the execution of the given Witnet Radon script in expectation that it will be relayed and solved by the Witnet DON.
+    /// A reward amount is escrowed by the Witnet Request Board that will be transfered to the reporter who relays back the Witnet-provided 
+    /// result to this request.
+    /// @dev Fails if:
+    /// @dev - provided reward is too low.
+    /// @dev - provided script is zero address.
+    /// @dev - provided script bytecode is empty.
+    /// @return _queryId The unique identifier of the posted request.
+    function postRequest(IWitnetRadon _script)
```

Comment:
As mentioned before, it is untrue that this is a Radon script and we should avoid calling it like that. Let's agree on a better terminology for once, because this is getting more an more difficult hahaha